packages/odyssey-react-mui/src/labs/SideNav/SideNav.tsx

```diff
@@ -402,31 +407,90 @@ const SideNav = ({
     [isCompact],
   );
 
-  const processedSideNavItems = useMemo(
-    () =>
-      sideNavItems.map((item) => ({
-        ...item,
-        children: item.children?.map((childProps) => {
-          return (
+  const setSelectedInChildItems = useCallback(
+    ({
+      nestedNavItems,
+      selectedItemId,
+    }: {
+      nestedNavItems: Array<Omit<SideNavItem, "startIcon" | "endIcon">>;
+      selectedItemId: string;
+    }) => {
+      const childItems = nestedNavItems.map((childItem) => {
+        if (childItem.isSelected) {
+          childItem.isSelected = false;
+        }
+        if (childItem.id === selectedItemId) {
+          childItem.isSelected = true;
+        }
+        if (childItem.nestedNavItems) {
+          setSelectedInChildItems({
+            nestedNavItems: childItem.nestedNavItems,
+            selectedItemId,
+          });
+        }
+        return childItem;
+      });
+      return childItems;
+    },
+    [],
+  );
+
+  const setSelectedItem = useCallback(
+    (selectedItemId: string) => {
+      const newNavItems = sideNavItemsList.map((item) => {
+        if (item.isSelected) {
+          item.isSelected = false;
+        }
+        if (item.id === selectedItemId) {
+          item.isSelected = true;
+        }
+        if (item.nestedNavItems) {
+          item.nestedNavItems = setSelectedInChildItems({
+            nestedNavItems: item.nestedNavItems,
+            selectedItemId,
+          });
+        }
+        return item;
+      });
+      updateSideNavItemsList(newNavItems);
+    },
+    [sideNavItemsList, setSelectedInChildItems],
+  );
+
+  const processedSideNavItems = useMemo(() => {
+    return sideNavItemsList?.map((item) => ({
+      ...item,
+      childNavItems: item.nestedNavItems?.map((childProps) => {
+        return {
+          id: childProps.id,
+          isSelected: childProps.isSelected,
+          isDisabled: childProps.isDisabled,
+          navItem: (
             <SideNavItemContentContext.Provider
-              value={{ ...sideNavItemContentProviderValue, depth: 2 }}
+              value={{
+                ...sideNavItemContentProviderValue,
+                depth: 2,
+                isSortable: item.isSortable,
+              }}
               key={childProps.id}
             >
               <SideNavItemContent
                 {...childProps}
                 key={childProps.id}
                 scrollRef={getRefIfThisIsFirstNodeWithIsSelected(childProps.id)}
+                onItemSelected={setSelectedItem}
               />
             </SideNavItemContentContext.Provider>
-          );
-        }),
-      })),
-    [
-      getRefIfThisIsFirstNodeWithIsSelected,
-      sideNavItems,
-      sideNavItemContentProviderValue,
-    ],
-  );
+          ),
+        };
+      }),
+    }));
```

Comment:
Super minor nit. This can be an implicit return. It will save on tabbing and make it easier to follow since it's just returning a value and no longer exists above it.

Because of the way we've built this, it's more confusing than it has to be in my opinion.

The whole `childNavItems` can be wrapped in `<SideNavItemContentContext.Provider>` right?

The depth calculation is the same for all sub-items. It doesn't need to wrap each one individually. If we were going to wrap each individually, we'd pass a prop.

This context is here to ensure we don't have to pass around props like that. My thinking is this _should_ be a context, but it needs to be in the render portion, not in the `useMemo` on each item.

On top of that, is `isSortable` a per-nav-item thing or the same for a group of nav items? If it's the same for that group, then we should add it to the provider around the group rather than each item.